test/test.hs

```diff
@@ -302,19 +302,14 @@ humanCommandM = do
     return . fst $ storeTrace (sactionPendingM tmp) [] -- log session
   abortOrCmd <- lift iterationM -- do stuff
   lift $ modify $ \sess ->
-    sess { sactionPendingM = dropEmpty $ sactionPendingM sess }
+    sess { sactionPendingM = dropEmptyBuffers $ sactionPendingM sess }
```

Comment:
Second fix: we want to remove all unnecessary action buffers at once.

Reply:
Ugh, that's a mess. Mistakenly I've pushed the file with unsaved changes. Both marked lines are doing the same thing. Sorry for inconvenience.

Reply:
No problem. :)

Reply:
Also, feel free to `push -f`. This is not master branch and I'm currently not working on that branch.

Reply:
I haven't quite wrapped my head yet around this new mock code, but I can see it matches the real code more closely, which is a definite value, even if some things never fire, etc. Better to keep it visually similar to the real code, as it is now, than to simplify it, removing the resemblance.